Fix scroll jerking

RobertBColton opened the pull request:
### Description of the Change

I originally reported the issue in #1165 and subsequently implemented the fix.

The scroll jerking is caused by centering around the item. We only want to scroll just enough to make the item visible, not center around it, otherwise keyboard navigation becomes awkward.

VS Code works this way using custom code in place of scrollIntoViewIfNeeded, because that is a Chrome only function. VS Code is based off of VS Team Services IDE which supports non-webkit browsers, such as Firefox, which don't have scrollIntoViewIfNeeded.

You could say that centering the scroll around the item leads to quicker keyboard navigation. I, however, do believe this is incorrect because if a keyboard user just wants to move the view a few items up it shouldn't move the whole view so much. It makes it easy to loose your place in the tree. That's the part that makes it awkward and probably why most GUI frameworks with tree/list widgets don't center the scroll with the keyboard navigation (I've tested Swing/Windows Forms/JavaFX).

### Alternate Designs

The only alternative I considered was making the center option a parameter of the scrollToEntry method. In a later commit (c39b672), I changed this PR to function this way.

### Benefits

Smoother and less awkward keyboard navigation of the tree-view.

### Possible Drawbacks

I have addressed the original concern that this would not be appropriate in all situations.

### Applicable Issues

#1165 is the original issue I filed.

50Wliu commented:
Hey @RobertBColton, glad to see that you got things working.  However, it looks like your change caused a spec failure - could you please take a look, and maybe also add a new spec asserting that the view only scrolls as much as necessary?  Thanks! 🙇

RobertBColton commented:
Ok, I finally got the chance to look at this again. I thought about it and realized that it is more appropriate to center around the active item in all other cases, such as when a file is first opened from the command line and Atom is just starting up. So instead of modifying the specs, I simply moved center to a parameter and allowed up/down to override the default and scroll just enough to make the item visible. I think this should be good to go, it's working great on over here and all tests are passing.

50Wliu commented:
That sounds great.

50Wliu commented:
Ok, two more things before I think this is mergeable.
1. The optional `center` parameter should read `center=true` (no whitespace around the `=` sign)
2. Specs/extra assertions should be added to the `core:move-up` and `core:move-down` tests to make sure that the view isn't centering.

RobertBColton commented:
Alright, I've addressed both of those issues. I made the spec do basically what I did to demonstrate the bug, select the very last entry and move up to the top emulating the up arrow key. Along the way it makes sure that each time the viewport scrolls it doesn't center the item. The specs seem to be passing and I closely inspected the values tested before committing.

50Wliu commented:
Is there a reason you added the assertions to a spec testing `tree-view:reveal-active-file` rather than [`core:move-up`](https://github.com/RobertBColton/tree-view/blob/b7eb9b468382e0070bf9eb88a0e0a4e5009ff9a3/spec/tree-view-package-spec.coffee#L838)?

RobertBColton commented:
Yes, to make it simpler. The core:move-up existing spec does not create enough files, so I added it to the spec that does create enough files to actually test with a scrollable tree view. Not only is the spec I added it to more focused on revealing active files, it also keeps us from having to recreate 20 text files which kind of slows down running of the package specs. I think it's more sensible this way.

50Wliu commented:
I think I'd still prefer to have it in `core:move-up` so as not to conflate what the spec is actually testing, but I don't see that as a blocker to merging this PR.

Thanks for the contribution! :tada:
